File: src/main/java/tk/avabin/tdg/beans/dtos/RPGSessionDto.java
package tk.avabin.tdg.beans.dtos;

import lombok.Data;
import org.springframework.context.annotation.Scope;
import org.springframework.stereotype.Component;

import java.util.Set;

/**
 * Created by Avabin on 18.05.2017.
 */
@Data
@Component
@Scope("prototype")
public class RPGSessionDto {
    private int id;
    private String name;
    private UserDto gameMaster;
    private Set<CharacterDto> characters;
    private Set<UserDto> players;
}

File: src/main/kotlin/tk/avabin/tdg/beans/dtos/RPGSessionDto.kt
package tk.avabin.tdg.beans.dtos

import org.springframework.context.annotation.Scope
import org.springframework.stereotype.Component
import java.io.Serializable

/**
 * Created by Avabin on 18.05.2017.
 */
@Component
@Scope("prototype")
data class RPGSessionDto(
        var id: Int = 0,
        var name: String = "",
        var gameMaster: UserDto? = null,
        var characters: Set<CharacterDto> = HashSet(),
        var players: Set<UserDto> = HashSet()
) : Serializable